Formula/mdt.rb

```diff
@@ -0,0 +1,16 @@
+class Mdt < Formula
+  desc "Command-line markdown todo list manager"
+  homepage "https://github.com/basilioss/mdt"
+  url "https://github.com/basilioss/mdt/archive/refs/tags/1.1.1.tar.gz"
+  sha256 "12bc8ec4c70487349a216b07182b88b9cb62d15816abae3a9f2b03207d5e019d"
+  license "GPL-3.0-or-later"
+  depends_on "gum"
+
+  def install
+    bin.install "mdt"
+  end
+
+  test do
+    system "#{bin}/mdt", "--version"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
It's fully interactive. The only non-interactive options are `--version` and `--help`. Should I replace `system "#{bin}/mdt", "--version"` with `assert_match "mdt 1.1.1", shell_output("#{bin}/mdt --version")`?

Reply:
We have a bunch of interactive tools that are tested more thoroughly. Checking the repo for usage of `PTY` should give some examples.

Reply:
Could you provide links to some examples? I've tried to check several TUI apps I know, but almost all of them use a simple `--version`/`--help` check:
- https://github.com/Homebrew/homebrew-core/blob/master/Formula/arttime.rb
- https://github.com/Homebrew/homebrew-core/blob/master/Formula/fff.rb
- https://github.com/Homebrew/homebrew-core/blob/master/Formula/cmatrix.rb
- https://github.com/Homebrew/homebrew-core/blob/master/Formula/neofetch.rb
- https://github.com/Homebrew/homebrew-core/blob/master/Formula/keepassc.rb

Would it be enough to also check for an invalid command as `bottom` [does](https://github.com/Homebrew/homebrew-core/blob/master/Formula/bottom.rb#L38)?